src/cls/rbd/cls_rbd.cc

```diff
@@ -5975,108 +5982,1206 @@ int mirror_remote_namespace_set(cls_method_context_t hctx, bufferlist *in,
   return 0;
 }
 
-/**
- * Input:
- * @param unique peer site name (std::string)
- * @param mirror_uuid (std::string)
- * @param direction (MirrorPeerDirection) -- future use
- *
- * Output:
- * @returns 0 on success, negative error code on failure
- */
-int mirror_peer_ping(cls_method_context_t hctx, bufferlist *in,
-                     bufferlist *out) {
-  std::string site_name;
-  std::string mirror_uuid;
-  cls::rbd::MirrorPeerDirection mirror_peer_direction;
-  try {
-    auto it = in->cbegin();
-    decode(site_name, it);
-    decode(mirror_uuid, it);
+int group_status_remove(cls_method_context_t hctx,
+			const string &global_group_id);
 
-    uint8_t direction;
-    decode(direction, it);
-    mirror_peer_direction = static_cast<cls::rbd::MirrorPeerDirection>(
-      direction);
-  } catch (const ceph::buffer::error &err) {
-    return -EINVAL;
+int group_remove(cls_method_context_t hctx, const string &group_id) {
+  bufferlist bl;
+  cls::rbd::MirrorGroup group;
+  int r = group_get(hctx, group_id, &group);
+  if (r < 0) {
+    if (r != -ENOENT) {
+      CLS_ERR("error reading mirrored group '%s': '%s'", group_id.c_str(),
+	      cpp_strerror(r).c_str());
+    }
+    return r;
   }
 
-  if (mirror_peer_direction != cls::rbd::MIRROR_PEER_DIRECTION_TX) {
-    return -EINVAL;
+  /* disable_group/group_remove at the time of force promote while the remote is alive
```

Comment:
@pkalever , does this still need to be commented out?

Reply:
Agree. I have uncommented this code in the proposed fixes , especially as the same code exists for images.